src/Microsoft.ML.Data/DataLoadSave/Binary/CodecFactory.cs

```diff
@@ -63,8 +63,7 @@ public CodecFactory(IHostEnvironment env, MemoryStreamPool memPool = null)
 
             // Register the old boolean reading codec.
             var oldBool = new OldBoolCodec(this);
-            RegisterOtherCodec(oldBool.LoadName, oldBool.GetCodec);
-
+            RegisterOtherCodec("DvBool", oldBool.GetCodec);
```

Comment:
Its for backward compatibility, incase someone saves a dataset in IDV format with the old DvTypes and loads in ML.NET without DvTypes.

---
In reply to: [213403080](https://github.com/dotnet/machinelearning/pull/692#discussion_r213403080) [](ancestors = 213403080)

Reply:
We aren't doing that for the other Dv types.... Why are we doing it for DvBool?

---
In reply to: [213408068](https://github.com/dotnet/machinelearning/pull/692#discussion_r213408068) [](ancestors = 213408068,213403080)

Reply:
I'm doing for DvDate/Time as well. For DvInt we don't need to change because the key for codec has not changed. The key is used to retrieve the codec to parse when the loader reads the signature string. 

DvBool is the only codec where we have also modified the codec as well to use one bit to write a boolean value as opposed to two bits. We need the old codec to read values from IDV that was written using DvBool.

---
In reply to: [213420305](https://github.com/dotnet/machinelearning/pull/692#discussion_r213420305) [](ancestors = 213420305,213408068,213403080)